core/src/main/scala/cats/instances/function.scala

```diff
@@ -9,7 +9,7 @@ import annotation.tailrec
 
 trait FunctionInstances extends cats.kernel.instances.FunctionInstances with Function0Instances with Function1Instances
 
-trait FunctionInstancesBinCompat0 {
+private[instances] trait FunctionInstancesBinCompat0 {
```

Comment:
I am not sure about introducing a BinCompat trait to fix a typo. The plan is to branch out scala 2.11 code right after 2.0 release.  Which means we shall be able to leave the typo in the 2.11 code and fix it in master branch without an extra bincompat trait.

Reply:
I can pull these parts out, but in my view it'd be better to get all the instance name-related source compat breakage out of the way at once, instead of spreading it across 2.0 and 2.1.

Reply:
The source breakage is highly unlikely to affect anyone (and if anyone it would be a super quick fix on compilation time), but the extra trait will be left there for a very long time.

Reply:
So you're saying that breaking source compat in 2.x releases is acceptable?